### ISSUE TITLE
SpaceFM donesn't follow the XDG thumbnail spec

### ISSUE THREAD

**Issue by sammoth** [IgnorantGuru/spacefm#642]
SpaceFM seems to store all its thumbnails in the ~/.thumbnails/normal folder regardless of the size they are generated at. The thumbnail cache has a spec that states what size thumbs should go into each .thumbnails subfolder:

https://specifications.freedesktop.org/thumbnail-spec/thumbnail-spec-latest.html#DIRSTRUCTURE

This would be useful to support to play nicer with other file managers so they can better share the cache.

At the moment I'm using

```
diff a/src/vfs/vfs-thumbnail-loader.c b/src/vfs/vfs-thumbnail-loader.c
@@ -417,9 +417,24 @@ static GdkPixbuf* _vfs_thumbnail_load( const char* file_path, const char* uri,
 #endif
     strcpy( ( file_name + 32 ), ".png" );

-    thumbnail_file = g_build_filename( g_get_home_dir(),
-                                       ".thumbnails/normal",
-                                       file_name, NULL );
+    if ( create_size <= 128 )
+    {
+        thumbnail_file = g_build_filename( g_get_home_dir(),
+                                           ".thumbnails/normal",
+                                        file_name, NULL );
+    }
+    else if ( create_size <= 256 )
+    {
+        thumbnail_file = g_build_filename( g_get_home_dir(),
+                                           ".thumbnails/large",
+                                        file_name, NULL );
+    }
+    else
+    {
+        thumbnail_file = g_build_filename( g_get_home_dir(),
+                                           ".thumbnails/huge",
+                                        file_name, NULL );
+    }

     if( G_UNLIKELY( 0 == mtime ) )
     {
```

**Comment by IgnorantGuru** [comment]
This does need to be corrected, although I don't see "huge" in the spec.  Maybe large can be used for >= 256?

**Comment by sammoth** [comment]
No you're quite right, sorry. I added that myself because I prefer the larger thumbnails being cached. I believe the way other file managers handle this is to not cache thumbnails larger than 256px and just regenerate them every time.

**Comment by IgnorantGuru** [comment]
Might take care of this too:  *If a file manager is aware of this standard and deletes a file it could take care of deleting the thumbnail too.*

**Comment by sroracle** [comment]
It would also be preferable to default to `$XDG_CACHE_HOME/thumbnails` before `$HOME/.thumbnails`, also per [the spec](https://specifications.freedesktop.org/thumbnail-spec/thumbnail-spec-latest.html#DIRECTORY).

**Comment by IgnorantGuru** [comment]
This has been fixed in alpha, due for 1.1.0 release.  SpaceFM will now:

* use $XDG_CACHE_HOME/thumbnails/ instead of ~/.thumbnails/
* use thumbnails/large/ for thumbnails > 128 pixel
* delete thumbnails when deleting associated files (in most cases)

It's probably best to delete ~/.thumbnails/ when upgrading, as SpaceFM won't remove or use files in this directory anymore.
